sqlmesh/core/engine_adapter/spark.py

```diff
@@ -334,39 +329,41 @@ def _get_data_objects(
             for row in results  # type: ignore
         ]
 
+    @property
+    def _spark_semver(self) -> t.Tuple[int, int, int]:
+        major, minor, patch, *_ = map(int, self.spark.version.split("."))
+        return major, minor, patch
+
     def get_current_catalog(self) -> t.Optional[str]:
         # Spark 3.4+ API
         if self._use_spark_session:
-            return self.spark.catalog.currentCatalog()
+            major, minor, _ = self._spark_semver
+            if major >= 3:
+                if major > 3 or minor >= 4:
+                    return self.spark.catalog.currentCatalog()
+                else:
+                    return self.spark.conf.get("spark.sql.defaultCatalog", "spark_catalog")
+            else:
+                # We just assume it is the default catalog
+                return "spark_catalog"
         return super().get_current_catalog()
 
     def set_current_catalog(self, catalog_name: str) -> None:
         # Spark 3.4+ API
-        self.spark.catalog.setCurrentCatalog(catalog_name)
+        major, minor, _ = self._spark_semver
+        if major > 3 or (major == 3 and minor >= 4):
```

Comment:
Seems like we can simplify with `(major, minor) >= (3, 4)`. Or if we get rid of the patch version we can compare the tuple directly and not unpack it.

Reply:
nice forgot about that pattern which is used for Python version checking. Updated: https://github.com/TobikoData/sqlmesh/pull/2409/commits/2f702b731393a32a90f69012197458ff38fb6158